embassy-stm32/src/lib.rs

```diff
@@ -47,6 +47,13 @@ pub mod crc;
     flash_l0, flash_l1, flash_wl, flash_wb, flash_l4, flash_f3, flash_f4, flash_f7, flash_h7
 ))]
 pub mod flash;
+#[cfg(not(any(
```

Comment:
this is a bit strange. Perhaps it's better to remove the `cfg` here, and put them inside `flash/mod.rs`. So `flash/mod.rs` contains the always-available stuff (`FlashRegion`), and then there's a cfg-guarded `flash/common.rs` with all the stuff that's available only on supported chips (mostly what's now in `flash/mod.rs`.)